jwst/extract_1d/extract.py

```diff
@@ -1793,42 +1655,49 @@ def assign_polynomial_limits(self, verbose):
             if verbose:
-                log.debug("Converting extraction limits to [[%g], [%g]]",
-                          lower, upper)
+                log.debug(f"Converting extraction limits to [[{lower}], [{upper}]]")
+
             self.p_src = [[create_poly([lower]), create_poly([upper])]]
         else:
             # The source extraction can include more than one region.
-            n_lists = len(self.src_coeff)
-            if n_lists // 2 * 2 != n_lists:
-                raise RuntimeError("src_coeff must contain alternating lists "
-                                   "of lower and upper limits.")
-            self.p_src = []
-            expect_lower = True                         # toggled in loop
-            for coeff_list in self.src_coeff:
-                if expect_lower:
-                    lower = create_poly(coeff_list)
-                else:
-                    upper = create_poly(coeff_list)
-                    self.p_src.append([lower, upper])
-                expect_lower = not expect_lower
+            n_src_coeff = len(self.src_coeff)
+            if n_src_coeff // 2 * 2 != n_src_coeff:
+                raise RuntimeError("src_coeff must contain alternating lists of lower and upper limits.")
+
+            self.p_src = self._poly_per_region(self.src_coeff)
 
         if self.bkg_coeff is not None:
-            n_lists = len(self.bkg_coeff)
-            if n_lists // 2 * 2 != n_lists:
-                raise RuntimeError("bkg_coeff must contain alternating lists "
-                                   "of lower and upper limits.")
-            self.p_bkg = []
-            expect_lower = True                         # toggled in loop
-            for coeff_list in self.bkg_coeff:
-                if expect_lower:
-                    lower = create_poly(coeff_list)
-                else:
-                    upper = create_poly(coeff_list)
-                    self.p_bkg.append([lower, upper])
-                expect_lower = not expect_lower
+            n_bkg_coeff = len(self.bkg_coeff)
+            if n_bkg_coeff // 2 * 2 != n_bkg_coeff:
+                raise RuntimeError("bkg_coeff must contain alternating lists of lower and upper limits.")
+
+            self.p_bkg = self._poly_per_region(self.bkg_coeff)
+
+    def _poly_per_region(self, coeffs):
+        result = []
+        expect_lower = True  # toggled in loop
+
+        for coeff_list in coeffs:
+            if expect_lower:
+                lower = create_poly(coeff_list)
+            else:
+                upper = create_poly(coeff_list)
+                self.p_src.append([lower, upper])
+
+            expect_lower = not expect_lower
+
+        return result
 
-    def extract(self, data, wl_array, verbose):
+    def extract(
+            self,
+            data: np.ndarray,
+            wl_array: Union[np.ndarray, None],
+            verbose: bool
+    ) -> Tuple[
+        float, float, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray
+    ]:
```

Comment:
In cases where you want to annotate a return that's super long (like this) it is possible to define your own type beforehand: 

```python
EXRACT_RESULTS = Tuple[
          float, float, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray
      ]
def extract(
    self, data: np.ndarray, wl_array: Union[np.ndarray, None], verbose: bool
)
```
This can be useful especially if several functions/methods return a similarly complex type. 

However, I personally would use this sparingly as it might add an unnecessary layer of obscurity.